### ISSUE TITLE
[BUG] Falco fails on Fastly's own generated_vcl VCL

### ISSUE THREAD

**Issue by alex8bitw** [ysugimoto/falco#337]
Please fill all **required** sections

### Kind of proposals

**required**
You can turn on multiple checkboxes.

- [] Parser
- [X] Linter
- [] Simulator
- [] Testing

### Describe the problem
I created a generated_vcl via API from a working config that had no errors.
I removed the "pragma" commands as those gave errors as well.

Running Falco gives errors like:
[ERROR] Type mismatch: bereq.http.connection requires type NULL but STRING was assigned (operator/assignment)
[ERROR] Type mismatch: bereq.http.X-Forwarded-For requires type NULL but STRING was assigned (operator/assignment)

also scope issues:
[ERROR] Variable "bereq.http.X-Forwarded-For" could not access in scope of RECV

There should be no ERRORS from Fastly's own generated VCL.

### VCL code that cause the problem / reproduceable
Use https://www.fastly.com/documentation/reference/api/vcl-services/vcl/#get-custom-vcl-generated and run Falco on the output.

### Expected behavior
No errors.

### Desktop (please complete the following information)
Falco 1.9.1
**required**
 - OS: Ubuntu 22.04.4 LTS  (WSL2)
 - Shell: bash,

**Comment by ysugimoto** [comment]
@alex8bitw Hi, thank you for reaching out to our project.

I'm guessing that this issue comes from VCL lines which is extracted Fastly boiler-plate macro.
If the lines of error could not present in your custom VCL, it's worth to try to enforce the subroutine scope by configuration.

Please try it refered to https://github.com/ysugimoto/falco/pull/296 (currentry this is undocumented for now, sorry)

```yaml
...
linter:
  enforce_subroutine_scopes:
    some_subroutine_name: ["recv"]
...
```

**Comment by ysugimoto** [comment]
Ah I've found that it's caused by `vcl_pipe` subroutine.
The `vcl_pipe` is undocumented and probably it's Varnish feature so falco could not recognize it.
Therefore, you could pass the linter configuration like:

```vcl
linter:
  ignore_subroutines:
    - vcl_pipe
```

**Comment by ysugimoto** [comment]
Your problems are fixed on #338, bear with us for releasing the latest version.
